### PR TITLE
Added feathered plane sample.

### PR DESCRIPTION
**Pull request by johnsietsma**
Create a feathered edge using UV mapping.
Demonstrates the use of ARPlaneMeshVisualizer.meshUpdated event to modify the generated mesh.
This doesn't create rounded corners (that's still to do).
Relies on the unpublished fix for index generation in ARFoundation.

### CODE REVIEW

**Comment by johnsietsma** [comment]
It's probably overkill to have a whole new scene here just to hold the prefab. As we get more scenes we should probably drop it and have a list of sample prefabs and sample scenes. But right now it may be better to have a scene just so people know it's there.

**Comment by timm-unity** [comment]
Looks good. Once we get the next revision of `ARFoundation` published, then I think this should just replace the existing plane prefab.

**Comment by johnsietsma** [comment]
I've updated code to match the coding conventions. @tdmowrer  You mentioned no public vars in your Slack comment, but they seem ok by the convention? Anything else I've missed?

@DanMillerDev I added the tint. This still relies on an unpublished fix to ARFoundation, so requires a local copy to build.

**Comment by tdmowrer** [comment]
ARFoundation preview 17 was published today, so that should have the triangulation fix you were waiting for, @johnsietsma .

File: Assets/Scripts/ARFeatheredPlaneMeshVisualizer.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Experimental.XR;
+using UnityEngine.XR.ARFoundation;
+
+[RequireComponent(typeof(ARPlaneMeshVisualizer), typeof(MeshRenderer), typeof(ARPlane))]
+public class ARFeatheredPlaneMeshVisualizer : MonoBehaviour
+{
+    [Tooltip("The width of the texture feathering (in world units).")]
+    public float m_FeatheringWidth = 0.2f;
```
Comment (tdmowrer) [inline Assets/Scripts/ARFeatheredPlaneMeshVisualizer.cs]: > You mentioned no public vars in your Slack comment...

Yeah, no public *fields*, only public *properties*. So

```csharp
public float m_FeatheringWidth = 0.2f;
```

should become

```csharp
[SerializeField]
float m_FeatheringWidth = 0.2f;

public float featheringWidth
{
    get { return m_FeatheringWidth; }
    set { m_FeatheringWidth = value; }
}
```

Our coding conventions say this
> Never expose public fields which are not const or static readonly. These fields should be published through a property.

and I've confirmed / been corrected by folks on the scripting team.

Comment (johnsietsma) [reply Assets/Scripts/ARFeatheredPlaneMeshVisualizer.cs]: Thanks. I've fixed this up.

**Comment by johnsietsma** [comment]
@tdmowrer Are you happy to accept the PR now?

**Comment by tdmowrer** [comment]
I have super minor coding standard nitpicks, but I don't want to block the merge any longer.

**Pull request merged**